npm/src/db/sql/sql.ts

```diff
@@ -140,27 +140,55 @@ class Sql implements DatabaseDriver {
     return JSON.parse(JSON.stringify(response)) || [];
   }
 
-  async getByIndex(namespace: string, idx: Index): Promise<any> {
-    const res = await this.indexRepository.findBy({
+  async getByIndex(namespace: string, idx: Index, pageOffset?: number, pageLimit?: number): Promise<any> {
+    const offsetAndLimitValueCheck = !dbutils.isNumeric(pageOffset) && !dbutils.isNumeric(pageLimit);
+    const skip = Number(offsetAndLimitValueCheck ? 0 : pageOffset);
+
+    let take = Number(offsetAndLimitValueCheck ? this.options.pageLimit : pageLimit);
+    let count = 0;
+
+    take += skip;
+    let res = await this.indexRepository.findBy({
```

Comment:
Can we not limit the sql query here instead of getting all records and then limiting it?